lib/internal/modules/cjs/helpers.js

```diff
@@ -117,7 +117,13 @@ function makeRequireFunction(mod, redirects) {
 
   resolve.paths = paths;
 
-  require.main = process.mainModule;
+  ObjectDefineProperty(require, 'main', {
+    __proto__: null,
+    configurable: true,
+    enumerable: true,
+    value: process.mainModule,
+    writable: true,
+  });
```

Comment:
We'll have a few properties that need to be defined like this and they all have the same descriptors aside from value. Might it be worthwhile to move this into a const and spread it in order to DRY things up a bit?

```js
  const pjsonFieldConfig = ObjectAssign(ObjectCreate(null), {
    __proto__: null,
    configurable: true,
    enumerable: true,
    writable: true,
  });
```